### ISSUE TITLE
Accessing initializationOptions in Nova Editor

### ISSUE THREAD

**Issue by tommasongr** [castwide/solargraph#436]
Hi there. I'm working on improving the Solargraph support in Nova editor. Unfortunately I'm experiencing some issues with the initializationOptions.

This is the object I'm passing as test with the client options. As you can see I disabled the hover and completion features, but the settings are not registered correctly.
```js
const clientOptions = {
            syntaxes: ["ruby"],
            initializationOptions: {
                hover: false,
                completion: false,
                logLevel: "info",
            },
};
```

The logLevel is apperently working fine. In fact I have some logs that show a response to the textDocument/hover and textDocument/completion.

```
Ruby Language Server[12:18:21.325000] [INFO] Sending response to textDocument/completion


Ruby Language Server[12:18:21.381000] [INFO] Server received textDocument/didChange


Ruby Language Server[12:18:21.382000] [INFO] Server received textDocument/completion

[...]

Ruby Language Server[12:27:45.289000] [INFO] Server received textDocument/hover


Ruby Language Server[12:27:45.350000] [INFO] Cataloging generic workspace


Ruby Language Server[12:27:45.375000] [INFO] Catalog complete (1 files, 2895 pins)


Ruby Language Server[12:27:45.411000] [INFO] Sending response to textDocument/hover
```

---

In the VSCode implementation I've seen another object passed in the client options:
```js
synchronize: {
         // Synchronize the setting section 'solargraph' to the server
	configurationSection: 'solargraph',
},
```

I supposed that may be related but unfortunately Nova doesn't allow to pass a synchronize object to the clientOptions yet. Writing the object in this way doesn't work either.
```js
initializationOptions: {
                solargraph: {
				hover: false,
				completion: false,
				logLevel: "info",
		}
},
```

I'm using `Solargraph 0.40.4`

What can I do? Thanks!

**Comment by castwide** [comment]
This appears to be an oversight in how the server gets initialized. It registers all capabilities when it starts, and only turns them off in response to a `workspace/didChangeConfiguration` message. I'll fix it in an upcoming patch release.

**Comment by castwide** [comment]
Fixed in 0.42.0.
